src/index.js

```diff
@@ -146,7 +147,7 @@ class Bitswap {
   _putBlock (block, callback) {
     this.blockstore.put(block, (err) => {
       if (err) {
-        return callback(err)
+        return setImmediate(() => callback(err))
```

Comment:
I tried to learn about `setImmediate()` and found https://stackoverflow.com/questions/47184855/when-is-setimmediate-required-in-node-callback

If that description is accurate, you wouldn't need a `setImmedate()` here as the callback comes from the outer function.

But I certainly lack a good understanding, so I might as well be wrong.

For the same cases I'll just comment "not sure if needed".